dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/ReconnectTimerTask.java

```diff
@@ -38,24 +38,26 @@
 
     @Override
     protected void doTask(Channel channel) {
-        try {
-            Long lastRead = lastRead(channel);
-            Long now = now();
-            if (lastRead != null && now - lastRead > heartbeatTimeout) {
-                logger.warn("Close channel " + channel + ", because heartbeat read idle time out: "
-                        + heartbeatTimeout + "ms");
-                if (channel instanceof Client) {
-                    try {
-                        ((Client) channel).reconnect();
-                    } catch (Exception e) {
-                        //do nothing
-                    }
-                } else {
+        Long lastRead = lastRead(channel);
+        Long now = now();
+        if (lastRead != null && now - lastRead > heartbeatTimeout) {
+            if (channel instanceof Client) {
+                try {
+                    logger.warn("Reconnect to remote channel " + channel.getRemoteAddress() + ", because heartbeat read idle time out: "
+                            + heartbeatTimeout + "ms");
+                    ((Client) channel).reconnect();
+                } catch (Throwable t) {
+                    logger.warn("Exception when reconnect to remote channel " + channel.getRemoteAddress(), t);
```

Comment:
Personally think that this place is worth discussing. Most of the exceptions have been handled in connect and disconnect, and there is a log. I think the log here can be removed.

Reply:
fixed